src/fhirtypes/InstanceDefinition.ts

```diff
@@ -26,8 +26,51 @@ export class InstanceDefinition {
   }
 
   toJSON(): any {
-    const clone = cloneDeep(this);
-    delete clone._instanceMeta; // Only needed for lookup and IG config - not a FHIR property
+    const orderedKeys: string[] = [];
+    const keys = Object.keys(this);
+    const underscoreKeys = remove(keys, key => key.startsWith('_'));
+    // Reconstruct the object with properties in the expected serialization order.
+    // resourceType, id, and meta are handled separately because they come before any other properties.
+    // When properties on a primitive field (e.g. "status"), exist, they will have
+    // an underscore-prefixed key (e.g. "_status"). If both of these properties exist
+    // on the instance, they should appear next to one another.
+    const clone: any = {
+      resourceType: cloneDeep(this.resourceType)
+    };
+    if (this._resourceType) {
+      clone._resourceType = cloneDeep(this._resourceType);
+    }
+    if (this.id) {
+      clone.id = cloneDeep(this.id);
+    }
+    if (this._id) {
+      clone._id = cloneDeep(this._id);
+    }
+    if (this.meta) {
+      clone.meta = cloneDeep(this.meta);
+    }
+    if (this._meta) {
+      clone._meta = cloneDeep(this._meta);
+    }
```

Comment:
I feel like this can be made more succinct in one of two ways.

1. You could just loop through a set of these keys:
```ts
const clone: any = {};
['resourceType', '_resourceType', 'id', '_id', 'meta', '_meta'].forEach(key => {
  if (this[key]) {
    clone[key] = cloneDeep(this[key]);
  }
});
```

2. Couldn't you just add them to `orderedKeys` first and then let the logic in lines 70-72 take care of it?
```ts
const orderedKeys = ['resourceType', '_resourceType', 'id', '_id', 'meta', '_meta']
  .filter(key => this[key] != null);
// _instanceMeta is only needed for lookup and IG config - not a FHIR property 
const keys = difference(Object.keys(this), [...orderedKeys, '_instanceMeta']);
const underscoreKeys = remove(keys, key => key.startsWith('_'));
```
I think something like the above 5 lines could replace the current lines 29 - 58.